addons/account/models/account_move.py

```diff
@@ -2287,6 +2287,9 @@ def copy(self, default=None):
         if copied_am.is_invoice(include_receipts=True):
             # Make sure to recompute payment terms. This could be necessary if the date is different for example.
             # Also, this is necessary when creating a credit note because the current invoice is copied.
+            if copied_am.currency_id.id != self.company_id.currency_id.id:
```

Comment:
```suggestion
            if copied_am.currency_id != self.company_id.currency_id:
```

Reply:
Done, thanks. But why is recordset equivalence better here?

Reply:
It's doing the same thing; just more elegant and overall clearer ;)